src/MSIdentityScaffolding/Microsoft.DotNet.MSIdentity/CodeReaderWriter/CodeWriter.cs

```diff
@@ -90,8 +90,65 @@ private static void InitUserSecrets(string projectPath)
             
             if (result.ExitCode != 0)
             {
+                Console.Write("FAILED\n");
                 throw new Exception("Error while running dotnet-user-secrets init");
             }
+            else
+            {
+                Console.Write("SUCCESS\n");
+            }
+        }
+
+        public static void AddPackage(string packageName, string packageVersion, string tfm)
+        {
+            if (!string.IsNullOrEmpty(packageName) && ((!string.IsNullOrEmpty(packageVersion)) || (!string.IsNullOrEmpty(tfm))))
+            {
+                var errors = new List<string>();
+                var output = new List<string>();
+                var arguments = new List<string>();
+                arguments.Add("package");
+                arguments.Add(packageName);
+                if (!string.IsNullOrEmpty(packageVersion))
+                {
+                    arguments.Add("-v");
+                    arguments.Add(packageVersion);
+                }
+
+                if (IsTfmPreRelease(tfm))
+                {
+                    arguments.Add("--prerelease");
+                }
+
+                if (!string.IsNullOrEmpty(tfm))
+                {
+                    arguments.Add("-f");
+                    arguments.Add(tfm);
+                }
+
+                Console.Write($"\nAdding package {packageName} . . . ");
+
+                var result = Command.CreateDotNet(
+                    "add",
+                    arguments.ToArray())
+                    .OnErrorLine(e => errors.Add(e))
+                    .OnOutputLine(o => output.Add(o))
+                    .Execute();
+
+                if (result.ExitCode != 0)
+                {
+                    Console.Write("FAILED\n");
+                    Console.WriteLine($"Failed to add package {packageName}");
+                }
+                else
+                {
+                    Console.Write("SUCCESS\n");
+                }
+            }
+        }
+
+        private static bool IsTfmPreRelease(string tfm)
+        {
+            return tfm.Equals("net6.0", StringComparison.OrdinalIgnoreCase);
```

Comment:
Shouldn't this do semver parsing to see if there any non-numerical parts?

Reply:
yup would be doing semver parsing, this was just a temporary check for the short tfm.  Tracking #1558